src/catching_show.c

```diff
@@ -0,0 +1,327 @@
+    GameStats *stats = Save_GameStats_Get(fieldSystem->saveData);
+    s64 elapsedTime = GF_RTC_TimeDelta(catchingShow->startTime, GF_RTC_DateTimeToSec());
+
+    if (elapsedTime < MAX_TIME_SECONDS) {
+        catchingShow->timePoints = ((MAX_TIME_SECONDS - elapsedTime) * POINTS_LOST_PER_SECOND);
+    } else {
+        catchingShow->timePoints = 0;
+    }
+
+    GameStats_AddScore(stats, SCORE_EVENT_17);
+}
+
+BOOL CatchingShow_CheckWildEncounter(FieldSystem *fieldSystem, int playerX, int playerY) {
+    if (IsStepCountZero(&sCatchingShow) == TRUE) {
+        return TryStartEncounter(fieldSystem, &sCatchingShow, playerX, playerY);
+    }
+    return FALSE;
+}
+
+BattleSetup *CatchingShow_GetBattleDTO(FieldSystem *fieldSystem) {
+    return SetupEncounter(fieldSystem, &sCatchingShow);
+}
+
+void CatchingShow_UpdateBattleResult(FieldSystem *fieldSystem, BattleSetup *setup) {
+    return UpdateBattleResultInternal(fieldSystem, setup, &sCatchingShow);
+}
+
+int CatchingShow_GetParkBallCount(FieldSystem *fieldSystem) {
+    return CATCHING_SHOW_MONS - NumMonsCaptured(&sCatchingShow);
+}
+
+int CatchingShow_CalcCatchingPoints(FieldSystem *fieldSystem) {
+    return CalcCatchingPoints(&sCatchingShow);
+}
+
+int CatchingShow_GetTimePoints(FieldSystem *fieldSystem) {
+    return GetTimePoints(&sCatchingShow);
+}
+
+u32 CatchingShow_GetTypePoints(FieldSystem *fieldSystem) {
+    return CalculateTypePoints(&sCatchingShow);
+}
+
+// Local functions
+
+static void BufferSpeciesData(u16 species, u8 *dest) {
+    GF_ASSERT(species != SPECIES_NONE && species <= NATIONAL_DEX_COUNT);
+    ReadFromNarcMemberByIdPair(dest, NARC_arc_ppark, 0, (species - 1) * 6, 6);
+}
+
+static void InitSpeciesData(FieldSystem *fieldSystem, CatchingShow *catchingShow) {
+    PalParkTransfer *transferData = Save_MigratedPokemon_Get(fieldSystem->saveData);
+    Pokemon *mon = AllocMonZeroed(HEAP_ID_4);
+    u8 narc_data[6];
+    u16 species;
+
+    for (int i = 0; i < CATCHING_SHOW_MONS; i++) {
+        catchingShow->caughtMonsOrder[i] = 0;
+        TransferDataToMon(transferData, i, mon);
+
+        species = GetMonData(mon, MON_DATA_SPECIES, NULL);
+
+        catchingShow->pokemon[i].species = species;
+        BufferSpeciesData(species, narc_data);
+
+        if (narc_data[0] != 0) {
+            catchingShow->pokemon[i].area = narc_data[PPMONDAT_OFFSET_LAND_SECTOR];
+        } else {
+            catchingShow->pokemon[i].area = narc_data[PPMONDAT_OFFSET_WATER_SECTOR] + (int)(PP_ENCTYPE_WATER_MIN - PP_ENCTYPE_LAND_MIN);
+        }
+        catchingShow->pokemon[i].rarity = narc_data[PPMONDAT_OFFSET_ENCOUTER_RATE];
+        catchingShow->pokemon[i].catchingPoints = narc_data[PPMONDAT_OFFSET_SCORE];
+        catchingShow->pokemon[i].type1 = GetMonData(mon, MON_DATA_TYPE_1, NULL);
+        catchingShow->pokemon[i].type2 = GetMonData(mon, MON_DATA_TYPE_2, NULL);
+    }
+
+    FreeToHeap(mon);
+}
+
+static int NumMonsCaptured(CatchingShow *catchingShow) {
```

Comment:
Naming:
Others might disagree (and if so then it's fine and there's no need to change it), but I prefer prefixes on internal functions as well